File: Task_3/task_3.cpp
#include <iostream>
#include <vector>
#include <algorithm> // заголовочный файл для reverse

using namespace std;

bool isNumberIncreasing(int number){ // функция для проверки числа на возрастание цифр
    if(number / 10 < 1){
        return false;
    } 
    vector<int> digits;

    while(number > 0){
        digits.push_back(number % 10);
        number /= 10;
    }

    reverse(digits.begin(), digits.end()); // переворачиваем вектор по итераторам

    for(size_t i = 1; i < digits.size(); i++){
        if(digits[i] <= digits[ i - 1]){
            return false;
        }
    }

    return true;
}

int reverseNumber(int number){ // функция для переворачивания числа
    vector<int> digits;

    while(number > 0){
        digits.push_back(number % 10);
        number /= 10;
    }

    int Number = 0;
    for(int digit : digits){
        Number = Number * 10 + digit;
    }

    return Number;
}

int main(){

    cout << "Введите числа, которые хотите проверить через пробел (CTRL + D для окончания): ";
    vector<int> numbers;

    int number; 
    while(cin >> number){ // ввод чисел, пока не будет введен EOF (CTRL + D)       
        if (number == 0) break;
        numbers.push_back(number);
    }

    cout << endl;
    short countSuccess = 0;
    for(int n : numbers){ // проверка чисел и вывод
        if(isNumberIncreasing(n)){
            cout << reverseNumber(n) << " ";
            countSuccess++;
        }
    }

    if(countSuccess == 0){
        cout << "Искомых чисел не найдено!" << endl;
    }

    return 0;
}

File: Task_3/task_3.ts
function isNumberIncreasing(number: number): boolean {
    if (Math.floor(number / 10) < 1) return false;
    const digits: number[] = [];

    let n = number;
    while (n > 0) {
        digits.push(n % 10);
        n = Math.floor(n / 10);
    }

    digits.reverse();
    for (let i = 1; i < digits.length; i++) {
        if (digits[i] <= digits[i - 1]) return false;
    }

    return true;
}

function reverseNumber(number: number): number {
    const digits: number[] = [];
    let n = number;
    while (n > 0) {
        digits.push(n % 10);
        n = Math.floor(n / 10);
    }

    let reversedNumber = 0;
    for (const digit of digits) {
        reversedNumber = reversedNumber * 10 + digit;
    }
    return reversedNumber;
}

function main() {
    const input = prompt("Введите числа, которые хотите проверить через пробел:");
    if (!input) {
        alert("Вы не ввели данные!");
        return;
    }

    const numbers = input
        .split(" ")
        .map((numStr) => parseInt(numStr, 10))
        .filter((num) => !isNaN(num));

    let result = "";
    let countSuccess = 0;

    for (const n of numbers) {
        if (isNumberIncreasing(n)) {
            result += `${reverseNumber(n)} `;
            countSuccess++;
        }
    }

    if (countSuccess === 0) {
        alert("Искомых чисел не найдено!");
    } else {
        alert(`Результат: ${result.trim()}`);
    }
}

// Запускаем программу
main();